### PR TITLE
pd: support pd microservices

### PR DESCRIPTION
**Pull request by HuSharp**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):https://github.com/pingcap/docs-tidb-operator/pull/2478

### CODE REVIEW

File: tiup/tiup-playground.md [hunk 1]
```diff
@@ -190,3 +190,16 @@ tiup playground scale-out --db 2
 ```shell
 tiup playground scale-in --pid 86526
 ```
+
+## PD 微服务部署
+
+正如 [PD 微服务](/pd-configuration-file.md#微服务部署) 中介绍，我们在 playground 中提供了简易的方式进行部署。
+
+- 需要指定 `--pd.mode` 为 `ms`，启动 PD 微服务模式需要指定参数 `--pd.api num`(num>=1)
+- 目前支持 TSO, Scheduling, resource manager 三个微服务，可以通过 `--pd.tso num`，`--pd.scheduling num`，`--pd.rm num` 来分别指定启动的实例数量。
```
Comment (rleungx) [inline tiup/tiup-playground.md:199]: rm 先不加怎么样，没用过

File: pd-configuration-file.md [hunk 2]
```diff
@@ -516,3 +516,38 @@ PD 中内置的 [Resource Control](/tidb-resource-control.md) 相关的配置项
...
+
+- backend-endpoints 需要传入以 API Mode 启动后的 PD 服务器地址。
+- listen-addr 为 TSO 服务的连接地址。
+- advertise-listen-addr 如无特殊需要，不指定或与 listen-addr 保持一致即可。
+
+其他诸如 TLS 证书等参数没有特殊之处，按照正常情况指定即可。
+
+> **注意：**
+>
+> 当前支持的微服务包括：TSO，Scheduling，Resource Manager。
```
Comment (rleungx) [inline pd-configuration-file.md:550]: ditto

File: pd-microservices.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+TiDB Operator 详细使用方法请参考以下文档：
+
+- [部署 PD 微服务](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/deploy-tidb-operator/configure-a-tidb-cluster.md#部署-pd-微服务)
+- [配置 PD 微服务](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/deploy-tidb-operator/configure-a-tidb-cluster.md#配置-pd-微服务)
+- [修改 PD 微服务](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/deploy-tidb-operator/modify-tidb-configuration.md#修改-pd-微服务配置)
+- [扩缩容 PD 微服务组件](https://docs.pingcap.com/zh/tidb-in-kubernetes/stable/deploy-tidb-operator/scale-a-tidb-cluster.md#扩缩容-pd-微服务组件)
+
+TiUP Playground 详细使用方法请参考以下文档：
+
+- [部署 PD 微服务](https://docs.pingcap.com/zh/tidb/stable/tiup/tiup-playground.md#pd-微服务部署)
```
Comment (qiancai) [inline pd-microservices.md:50]:
```suggestion
- [部署 PD 微服务](/tiup/tiup-playground.md#pd-微服务部署)
```

File: tiup/tiup-playground.md [hunk 4]
```diff
@@ -190,3 +190,16 @@ tiup playground scale-out --db 2
...
+## 部署 PD 微服务
+
+从 v8.0.0 起，PD 支持[微服务模式](/pd-microservices.md)。你可以通过 TiUP playground 为集群部署 `tso` 微服务和 `scheduling` 微服务。
+
+```shell
+./tiup-playground v8.0.0 --pd.mode ms --pd.api 3  --pd.tso 2 --pd.scheduling 3
+```
+
+- `--pd.mode`：当指定 `--pd.mode` 为 `ms` 时，代表启用 PD 微服务模式。
+- `--pd.api num`：指定 PD 微服务 API 的数量，需要大于等于 1。
```
Comment (qiancai) [inline tiup/tiup-playground.md:203]:
```suggestion
- `--pd.api num`：指定 PD 微服务 API 的数量，需要大于等于 `1`。
```

File: tiup/tiup-playground.md [hunk 5]
```diff
@@ -191,13 +191,15 @@ tiup playground scale-out --db 2
...
+从 v8.0.0 起，PD 支持[微服务模式](/pd-microservices.md)。你可以通过 playground 部署 PD 的 TSO 微服务和 Scheduling 微服务。
 
 ```shell
 ./tiup-playground v8.0.0 --pd.mode ms --pd.api 3  --pd.tso 2 --pd.scheduling 3
-```
\ No newline at end of file
+```
+
+- `--pd.mode`：当指定 `--pd.mode` 为 `ms` 时，代表启用 PD 微服务模式。
+- `--pd.api num`：指定 API 线程的数量，需要大于等于 1。
```
Comment (rleungx) [inline tiup/tiup-playground.md]: 下一个版本可能会继续使用 --pd.num 的方式替代，https://github.com/pingcap/tiup/pull/2381

File: pd-microservices.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+- `tso` 微服务目前不支持动态启停，开启或关闭 `tso` 微服务后你需要重启 PD 集群才会生效。
+- 只有 TiDB 组件支持通过服务发现直接连接 `tso` 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 `tso` 微服务以获取时间戳。
+- 与[同步部署模式 (DR Auto-Sync)](https://docs.pingcap.com/zh/tidb/stable/two-data-centers-in-one-city-deployment) 特性不兼容。
+- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
+- 由于集群中可能存在静默 Region，`scheduling` 微服务在进行主备切换时，为避免冗余调度，集群可能存在最多五分钟内没有调度的现象。
+
+## 使用方法
+
+目前 PD 微服务支持通过 TiDB Operator 和 TiUP Playground 进行部署。
```
Comment (lilin90) [inline pd-microservices.md:43]:
```suggestion
目前 PD 微服务支持通过 TiDB Operator 或 TiUP Playground 进行部署。
```

File: tiup/tiup-playground.md [hunk 7]
```diff
@@ -190,3 +190,16 @@ tiup playground scale-out --db 2
 ```shell
 tiup playground scale-in --pid 86526
 ```
+
+## 部署 PD 微服务
+
+从 v8.0.0 起，PD 支持[微服务模式](/pd-microservices.md)（实验特性）。你可以通过 TiUP playground 为集群部署 `tso` 微服务和 `scheduling` 微服务。
```
Comment (lilin90) [inline tiup/tiup-playground.md:196]:
```suggestion
从 v8.0.0 起，PD 支持[微服务模式](/pd-microservices.md)（实验特性）。你可以通过 TiUP Playground 为集群部署 `tso` 微服务和 `scheduling` 微服务。
```

File: pd-microservices.md [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+从 v8.0.0 开始，PD 支持微服务模式。该模式可将 PD 的时间戳分配和集群调度功能拆分为以下微服务单独部署，从而与 PD 的路由功能解耦，让 PD 专注于元数据的路由服务。
+
+- `tso` 微服务：为整个集群提供单调递增的时间戳分配。
+- `scheduling` 微服务：为整个集群提供调度功能，包括但不限于负载均衡、热点处理、副本修复、副本放置等。
+
+每个微服务都以独立进程的方式部署。当设置某个微服务的副本数大于 1 时，该微服务会自动实现主备的容灾模式，以确保服务的高可用性和可靠性。
+
+> **警告：**
+>
+> PD 微服务目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/pingcap/tidb/issues) 反馈。
```
Comment (HuSharp) [inline pd-microservices.md:17]:
```suggestion
> PD 微服务目前为实验特性，不建议在生产环境中使用。该功能可能会在未事先通知的情况下发生变化或删除。如果发现 bug，请在 GitHub 上提 [issue](https://github.com/tikv/pd/issues) 反馈。
```

**Review by qiancai — APPROVED** [review]

File: pd-microservices.md [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+> **注意：**
+>
+> 如果集群的性能瓶颈不是 PD 引起的，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+- `tso` 微服务目前不支持动态启停，开启或关闭 `tso` 微服务后你需要重启 PD 集群才会生效。
+- 只有 TiDB 组件支持通过服务发现直接连接 `tso` 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 `tso` 微服务以获取时间戳。
+- 与[同步部署模式 (DR Auto-Sync)](https://docs.pingcap.com/zh/tidb/stable/two-data-centers-in-one-city-deployment) 特性不兼容。
+- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](https://docs.pingcap.com/zh/tidb/stable/system-variables#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
```
Comment (qiancai) [inline pd-microservices.md:38]:
```suggestion
- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
```

File: pd-microservices.md [hunk 10]
```diff
@@ -0,0 +1,81 @@
...
+>
+> 如果集群的性能瓶颈不是 PD 引起的，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+- `tso` 微服务目前不支持动态启停，开启或关闭 `tso` 微服务后你需要重启 PD 集群才会生效。
+- 只有 TiDB 组件支持通过服务发现直接连接 `tso` 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 `tso` 微服务以获取时间戳。
+- 与[同步部署模式 (DR Auto-Sync)](https://docs.pingcap.com/zh/tidb/stable/two-data-centers-in-one-city-deployment) 特性不兼容。
+- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
+- 由于集群中可能存在静默 Region，`scheduling` 微服务在进行主备切换时，为避免冗余调度，集群可能存在最多五分钟内没有调度的现象。
```
Comment (lhy1024) [inline pd-microservices.md:39]: 需要指明系统配置？

**Review by lhy1024 — COMMENTED** [review]
the rest LGTM

File: pd-microservices.md (hunk 10 above)
Comment (HuSharp) [reply pd-microservices.md]: 已修改 https://github.com/pingcap/docs-cn/pull/15975/commits/a55d41a9c09e2e2cdf43142021ebbcc61af82138#diff-c3340841cc6bf8a5d0fdd646f8491802ad6459f18c62b9ec92966cb3bdcd9258R39

cc @qiancai

File: pd-microservices.md [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+>
+> 如果集群的性能瓶颈不是 PD 引起的，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+- `tso` 微服务目前不支持动态启停，开启或关闭 `tso` 微服务后你需要重启 PD 集群才会生效。
+- 只有 TiDB 组件支持通过服务发现直接连接 `tso` 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 `tso` 微服务以获取时间戳。
+- 与[同步部署模式 (DR Auto-Sync)](https://docs.pingcap.com/zh/tidb/stable/two-data-centers-in-one-city-deployment) 特性不兼容。
+- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
+- 由于集群中可能存在[静默 Region](https://docs.pingcap.com/zh/tidb/stable/tikv-configuration-file#hibernate-regions)，`scheduling` 微服务在进行主备切换时，为避免冗余调度，集群可能存在最多 `peer-stale-state-check-interval` 时间内（默认为五分钟）没有调度的现象。
```
Comment (qiancai) [inline pd-microservices.md:39]:
```suggestion
- 由于集群中可能存在[静默 Region](/tikv-configuration-file.md#hibernate-regions)，`scheduling` 微服务在进行主备切换时，为避免冗余调度，集群可能存在最多 `peer-stale-state-check-interval` 时间内（默认为五分钟）没有调度的现象。
```

Comment (HuSharp) [reply pd-microservices.md:39]: 上面 dr auto sync 模式引用的链接需要改不？

Comment (qiancai) [reply pd-microservices.md:39]: 是的，也需要修改

File: pd-microservices.md [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+
+> **注意：**
+>
+> 如果集群的性能瓶颈不是 PD 引起的，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+- `tso` 微服务目前不支持动态启停，开启或关闭 `tso` 微服务后你需要重启 PD 集群才会生效。
+- 只有 TiDB 组件支持通过服务发现直接连接 `tso` 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 `tso` 微服务以获取时间戳。
+- 与[同步部署模式 (DR Auto-Sync)](https://docs.pingcap.com/zh/tidb/stable/two-data-centers-in-one-city-deployment) 特性不兼容。
```
Comment (qiancai) [inline pd-microservices.md:37]:
```suggestion
- 与[同步部署模式 (DR Auto-Sync)](/two-data-centers-in-one-city-deployment.md) 特性不兼容。
```

**Review by lhy1024 — APPROVED** [review]

File: pd-microservices.md [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+>
+> 如果集群的性能瓶颈不是 PD 引起的，则无需开启微服务。微服务本身会增加组件数量，提高运维成本。
+
+## 使用限制
+
+- `tso` 微服务目前不支持动态启停，开启或关闭 `tso` 微服务后你需要重启 PD 集群才会生效。
+- 只有 TiDB 组件支持通过服务发现直接连接 `tso` 微服务，其他的组件是通过请求转发的方式，将请求通过 PD 转发到 `tso` 微服务以获取时间戳。
+- 与[同步部署模式 (DR Auto-Sync)](/two-data-centers-in-one-city-deployment.md) 特性不兼容。
+- 与 TiDB 系统变量 [`tidb_enable_tso_follower_proxy`](/system-variables.md#tidb_enable_tso_follower_proxy-从-v530-版本开始引入) 不兼容。
+- 由于集群中可能存在[静默 Region](/tikv-configuration-file.md#hibernate-regions)，`scheduling` 微服务在进行主备切换时，为避免冗余调度，集群可能存在最多 [`peer-stale-state-check-interval`](/tikv-configuration-file.md#peer-stale-state-check-interval) 时间内（默认为五分钟）没有调度的现象。
```
Comment (qiancai) [inline pd-microservices.md:39]:
```suggestion
- 由于集群中可能存在[静默 Region](/tikv-configuration-file.md#hibernate-regions)，`scheduling` 微服务在进行主备切换时，为避免冗余调度，集群可能存在最长 [`peer-stale-state-check-interval`](/tikv-configuration-file.md#peer-stale-state-check-interval) 时间内（默认为五分钟）没有调度的现象。
```

**Pull request merged**